### PR TITLE
abstract some OS operations

### PR DESCRIPTION
**Pull request by devoncarew**
- abstract some OS operations (`makeExecutable()` and `killTcpPortListeners()`)
- fix an exception under windows (seen here: https://github.com/flutter/tools/pull/161)

### CODE REVIEW

**Comment by TheBosZ** [comment]
+1

**Comment by devoncarew** [comment]
@abarth, @Hixie

**Comment by devoncarew** [comment]
also, update the `type` ==> `icon` change in the template app

File: lib/src/os_utils.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
+
```
Comment (abarth) [inline lib/src/os_utils.dart]: License block pls.

File: lib/src/os_utils.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import 'package:logging/logging.dart';
+
+import 'process.dart';
+
+final OperatingSystemUtils osUtils = new OperatingSystemUtils();
+
+final Logger _logging = new Logger('sky_tools.os');
+
+abstract class OperatingSystemUtils {
+  factory OperatingSystemUtils() {
```
Comment (abarth) [inline lib/src/os_utils.dart]: Maybe make the factory private?

**Comment by abarth** [comment]
LGTM

File: lib/src/os_utils.dart (hunk 1 above)
Comment (devoncarew) [inline lib/src/os_utils.dart]: added

File: lib/src/os_utils.dart (hunk 2 above)
Comment (devoncarew) [inline lib/src/os_utils.dart]: will do

**Comment by Hixie** [comment]
Travis is mad at you.

**Comment by TheBosZ** [comment]
Just tell him to calm down!

**Comment by devoncarew** [comment]
Yeah, I pushed this guy in:

https://github.com/flutter/tools/pull/162/files#diff-01ef21ce686e670d8f850c0d2fd5a9c8R164

but the published packages don't yet have that field change -

**Comment by devoncarew** [comment]
Sry, specifically `path` ==> `icon` (https://github.com/flutter/tools/blob/master/lib/src/commands/init.dart#L164).

**Comment by Hixie** [comment]
ah ok.

**Comment by TheBosZ** [comment]
Wasn't it ```type ==> icon```?

**Comment by devoncarew** [comment]
Ah yeah, `type`.

**Comment by abarth** [comment]
Publishing now.

**Pull request merged**